pkg/kubelet/pleg/generic.go

```diff
@@ -185,19 +185,20 @@ func (g *GenericPLEG) relist() {
 		metrics.PLEGRelistInterval.Observe(metrics.SinceInMicroseconds(lastRelistTime))
 	}
 
+	// Get all the pods.
+	podList, err := g.runtime.GetPods(true)
+	if err != nil {
+		glog.Errorf("GenericPLEG: Unable to retrieve pods: %v", err)
+		return
+	}
+
```

Comment:
You need to move `timestamp := g.clock.Now()` and the defer block for `PLEGRelistLatency` to  *before* the `GetPods`. Otherwise, we won't be recording the right metrics.